src/System.Private.CoreLib/src/System/Threading/ManagedThreadId.cs

```diff
@@ -13,17 +13,38 @@
 using System.Runtime;
 using System.Runtime.InteropServices;
 using System.Security;
-
-#pragma warning disable 0420
-
+using System.Collections.Generic;
 
 namespace System.Threading
 {
     internal class ManagedThreadId
     {
+        private class RecycledIdsList
+        {
+            internal RecycledIdsList(int id) { _id = id; }
+            internal int _lock;
+            internal int _id;
+            internal RecycledIdsList _next;
+        }
+
+        [ThreadStatic]
+        private static ManagedThreadId t_currentThreadId;
         [ThreadStatic]
         private static int t_currentManagedThreadId;
 
+        // We have to avoid the static constructors on the ManagedThreadId class, otherwise we can run into stack overflow as first time Current property get called, 
+        // the runtime will ensure running the static constructor and this process will call the Current property again (when taking any lock) 
+        //      SharedLibrary!System::Environment.get_CurrentManagedThreadId
+        //      SharedLibrary!System::Threading::Lock.Acquire
```

Comment:
I would remove the `SharedLibrary!` from this comment to make it easier to read.